sdks/java/core/src/main/java/org/apache/beam/sdk/values/Row.java

```diff
@@ -346,13 +347,48 @@ public boolean equals(Object o) {
       return false;
     }
     Row other = (Row) o;
-    return Objects.equals(getSchema(), other.getSchema())
-        && Objects.equals(getValues(), other.getValues());
+
```

Comment:
I believe that we can simply replace Objects.equals with Objects.deepEquals here instead of this code.